packages/@aws-cdk/aws-ecs/lib/fargate/fargate-task-definition.ts

```diff
@@ -0,0 +1,59 @@
+import cdk = require('@aws-cdk/cdk');
+import { BaseTaskDefinition, BaseTaskDefinitionProps, Compatibilities, NetworkMode } from '../base/base-task-definition';
+
+/**
+ * Properties to define a Fargate Task
+ */
+export interface FargateTaskDefinitionProps extends BaseTaskDefinitionProps {
```

Comment:
That doesn't seem like a good justification for splitting Fargate/EC2 for task definitions. Going back to the motivation - a task definition should be portable(-ish) between EC2 and Fargate. I would rather we add a couple of runtime checks in `TaskDefinition` and maintain a single class for both then get this tiny bit of compile-time check, but lose the portablity.

Reply:
I don't think it should be broken up--we don't break it up in the console even. One of the things that we talk about is how you can easily migrate from EC2 to Fargate and vice versa. Also if you use awsvpc you can easily create a task definition that works for both. There isn't any field in a task definition that makes it Fargate or EC2 specific--just the values or IAM roles you give--but both can take the same things.

Reply:
re: task def portability.  This is only sort of true.  A Fargate task definition can be used in EC2, but an EC2 task definition cannot necessarily be used in Fargate.  I would like the class structure to actually reflect that.  Otherwise it is difficult to understand what task def objects can be used in what service, because you have to know whether the 'capabilities' flag has been set inside of the task def object to use it for Fargate.

Reply:
Can typescript take in 1 or 2 values? E.g. give it Fargate or EC2 or Fargate and EC2 into a single task def function?

Reply:
> A Fargate task definition can be used in EC2

Also not necessarily true. In Fargate you can choose to omit per-container memory specifiers (only specify at the task level), but then the EC2 scheduler will yell at you that it needs at least a limit or a reservation. Right?

Reply:
lol yes you're right, our validation logic is so complicated even I forget some of the rules...